### ISSUE TITLE
build fails with edge rendering when accessing node built-ins in a prerendered route

### ISSUE THREAD

**Issue by mayank99** [withastro/astro#8016]
### What version of `astro` are you using?

2.10.4

### Are you using an SSR adapter? If so, which one?

Vercel

### What package manager are you using?

npm

### What operating system are you using?

Mac

### What browser are you using?

Firefox

### Describe the Bug

`astro build` fails when using `@astrojs/vercel/edge` and trying to import a node built-in (for example `promisify` or `fs`) in a prerendered route.

```astro
---
import { promisify } from 'node:util';

export const prerender = true;
---
```

> [ERROR] Could not resolve "node:util".
> The package "node:util" wasn't found on the file system but is built into node. Are you trying to bundle for node? You can use "platform: 'node'" to do that, which will remove this error.

### What's the expected result?

I understand that edge runtimes have restrictions (i believe Vercel edge uses some proprietary Cloudflare runtime, and Netlify uses Deno), however the code in pre-rendered routes is executed at build-time, so  Node built-ins should be usable without breaking build.

The code that needs to be executed for pre-rendering any routes should not be bundled into the build output.

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-u5dbiv-qk17ar?file=src%2Fpages%2Fprerendered.astro,astro.config.mjs

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by natemoo-re** [comment]
Hmm, interesting! I know Astro typically removes any prerendered routes from the bundle before passing it to the adapter, but it looks like that's not happening at the moment. We'll have to figure out what's going on.

**Comment by matthewp** [comment]
Hey @mayank99 thanks for the issue, but we've actually just dropped support for Vercel Edge adapter and are recommending people use Serverless + Edge Middleware instead. For this very sort of reason, compatibility is such a problem that few people are using these edge adapters in practice. Going to close as we won't be fixing Edge adapter issues going forward.

**Comment by mayank99** [comment]
so if I was to build my own adapter for edge rendering, would Astro still include the frontmatter code from prerendered route in the bundle that is handed off to me?

**Comment by matthewp** [comment]
The adapter would need to configure those modules to be external. This is how we've handled adapter special cases in the past, Astro core does not do so.

**Comment by lilnasy** [comment]
Fixed in #8070
